### PR TITLE
Improved HL commander - spiral & linear segment

### PR DESCRIPTION
**Pull request by matejkarasek**
This PR extends the functionality of the HL commander by adding 1) a linear option to the GOTO command and 2) a spiral command.
More details in the corresponding firmware PR: https://github.com/bitcraze/crazyflie-firmware/pull/1410

### CODE REVIEW

**Comment by ataffanel** [comment]
It seems that this PR will make the lib incompatible with previous version of the firmware.

Since the lib is currently not checking for the protocol version and locking connection, it needs to stay compatible with the old message. Either by checking the version and sending the appropriate message (maybe with a warning to update the Crazyflie), or by adding the new message in a new function.

**Comment by matejkarasek** [comment]
So I made a new function.
Checking for the may have been more elegant, however, I have not found a way how to do that...

**Comment by knmcguire** [comment]
I've just approved the CI run as that hasn't been done yet, so please check if there are any errors there. 

Also for the review we'll need to try this out ourselves just to make sure that everything still works, so we will try to do that before the next release.

Yes absolutely, that would be a nice feature. I'm not sure if we have a issue/ticket for the version checking yet. 🤔

**Comment by knmcguire** [comment]
The CI has failed on some autopep8 checks. @gemenerik will take a look at your PRs for test flight so if he can he'll fix it or you can fix it as well. 

You can also run the test locally on your computer with the toolbelt: https://www.bitcraze.io/documentation/repository/toolbelt/master/ and `tb verify`. It should work for windows as well but there are some ifs/buts for the instructions.

**Comment by matejkarasek** [comment]
Thanks, the failed checks should now be fixed.

I have tested this on the crazyflie with both the latest release as well as with the new firmware https://github.com/bitcraze/crazyflie-firmware/pull/1410. All works as expected. 

But indeed this should be tested by someone else before it gets pulled into master.

**Review by gemenerik — CHANGES_REQUESTED** [review]

File: cflib/crazyflie/high_level_commander.py [hunk 1]
```diff
@@ -151,6 +153,53 @@ def go_to(self, x, y, z, yaw, duration_s, relative=False,
                                       x, y, z,
                                       yaw,
                                       duration_s))
+        
+    def go_to2(self, x, y, z, yaw, duration_s, relative=False, linear=False,
```
[the remark this answers is not in the document]
Comment (matejkarasek) [reply cflib/crazyflie/high_level_commander.py:157]: Thanks for checking @gemenerik 
I was looking for that function, somehow I missed that :)

I will modify it as you suggest 👍

**Review by matejkarasek — COMMENTED** [review]
All should be fixed as requested. Succesfully flight tested

**Review by gemenerik — APPROVED** [review]

**Pull request merged**